docs/development/doc_guide.rst

```diff
@@ -1,76 +1,849 @@
+
+This reST_ block will render as:
+
+   An example of in-line math is :math:`x`. Using Unicode characters
+   like :math:`α + β + γ` makes math easier to read in the source code.
+
+Markdown
+--------
+
+A few of PlasmaPy's files are written using Markdown_, such as README
+files and licenses from other packages. Markdown_ is simpler but more
+limited than reST_. Markdown_ files end with :file:`.md`. Posts on
+GitHub are written in `GitHub Flavored Markdown`_.  The following code
+block contains a few common examples of Markdown_ formatting.
+
+.. code-block:: markdown
+
+   # Header 1
+
+   ## Header 2
+
+   Here is a link to [PlasmaPy's documentation](https://docs.plasmapy.org).
+
+   We can make text **bold** or *italic*.
+
+   We can write in-line code like `x = 1` or create a Python code block:
+
+   ```Python
+   y = 2
+   z = 3
+   ```
+
+Sphinx
+------
+
+Sphinx_ is the software used to generate `PlasmaPy's documentation`_
+from reST_ files and Python_ docstrings. It was originally created to
+write Python's documentation and has become the de facto software for
+documenting Python_ packages. Almost all Python_ open-source packages
+utilize Sphinx_ to generate their documentation.
+
+Configuration
+~~~~~~~~~~~~~
+
+The `docs/conf.py`_ file contains the configuration information needed
+to customize Sphinx_ behavior.  The documentation for Sphinx_ lists the
+`configuration options`_ that can be set.
+
+The `docs/_static/sphinx_rtd_overrides.css`_ file contains `style
+overrides`_ for the `Read the Docs Sphinx Theme`_ to customize the look
+and feel of the online documentation.
+
+Sphinx extensions
+~~~~~~~~~~~~~~~~~
+
+`PlasmaPy's documentation`_ is built with the following Sphinx_
+extensions:
+
+* `sphinx.ext.autodoc` for including documentation from docstrings
+* `sphinx.ext.intersphinx` for linking to other projects' documentation
+* `sphinx.ext.graphviz` to allow Graphviz_ graphs to be included
+* `sphinx.ext.mathjax` for math rendering with MathJax_
+* `sphinx.ext.napoleon` for allowing NumPy style docstrings
+* `sphinx.ext.todo` to support ``todo`` directives
+* nbsphinx_ for including Jupyter_ notebooks
+* `sphinx_copybutton`_ to add a "copy" button for code blocks
+* `sphinx_gallery.load_style`_ for using sphinx-gallery styles
+* `IPython.sphinxext.ipython_console_highlighting`_
+* `sphinx_changelog`_ for rendering `towncrier`_ changelogs
+* `plasmapy_sphinx` for customizations created for use in PlasmaPy and
+  affiliated packages. Note that `plasmapy_sphinx` is expected to be
+  broken out into its own package in the future.
+
+These extensions are specified in :confval:`extensions` configuration
+value in `docs/conf.py`_.
+
+.. _external-references:
+
+References to other packages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Intersphinx_ allows the automatic generation of links to the
+documentation of objects in other projects. This cross-package linking
+is made possible with the `sphinx.ext.intersphinx` extension.
```

Comment:
```suggestion
Intersphinx_ allows the automatic generation of links to the
documentation of objects in other projects. This cross-package linking
is made possible with the `sphinx.ext.intersphinx` extension and proper
package indexing by the external package using `sphinx.ext.autodoc`.
```

As you've discovered recently, proper cross-referencing is twofold: (1) us setting up `intersphinx` properly and (2) the desired package properly indexing their functionality.